### ISSUE TITLE
SQL parser reports wrong error.

### ISSUE THREAD

**Issue by den-crane** [ClickHouse/ClickHouse#19477]
```sql

CREATE TABLE items (
  time DateTime,
  group_id UInt16,
  value UInt32()
) ENGINE = MergeTree()
PARTITION BY toYYYYMM(time)
ORDER BY (group_id, time);

SELECT
  group_id,
  groupArray(time)[1] as time,
  groupArray(value)                       , 
FROM ( 
  SELECT * FROM items ORDER BY time desc LIMIT 100 BY group_id
)
GROUP BY group_id;


(version 21.1.1)
Received from localhost:9000. DB::Exception: Query argument can only be used in the `view` TableFunction.

```
because of excessive comma  ` groupArray(value)    , `

**Comment by alexey-milovidov** [comment]
It has parsed FROM as function.
Note that our parsed does not have keywords at all...

I can add a quirk to the parser, so FROM can never be parsed as a function.

**Comment by fm4v** [comment]
[Fixed in 23.4](https://fiddle.clickhouse.com/0d5663da-557d-4b74-98c0-1bd987a02dfe)
